components/tidb_query/src/rpn_expr/impl_json.rs

```diff
@@ -57,6 +136,127 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_json_array() {
+        let cases = vec![
+            (vec![], Some(r#"[]"#)),
+            (vec![Some(r#"1"#), None], Some(r#"[1, null]"#)),
+            (
+                vec![
+                    Some(r#"1"#),
+                    None,
+                    Some(r#"2"#),
+                    Some(r#""sdf""#),
+                    Some(r#""k1""#),
+                    Some(r#""v1""#),
+                ],
+                Some(r#"[1, null, 2, "sdf", "k1", "v1"]"#),
+            ),
+        ];
+
+        for (vargs, expected) in cases {
+            let vargs = vargs
+                .into_iter()
+                .map(|input| input.map(|s| Json::from_str(s).unwrap()))
+                .collect::<Vec<_>>();
+            let expected = expected.map(|s| Json::from_str(s).unwrap());
+
+            let output = RpnFnScalarEvaluator::new()
+                .push_params(vargs.clone())
+                .evaluate(ScalarFuncSig::JsonArraySig)
+                .unwrap();
+            assert_eq!(output, expected, "{:?}", vargs);
+        }
+    }
+
+    #[test]
+    fn test_json_merge() {
+        let cases = vec![
+            (vec![None, None], None),
+            (vec![Some("{}"), Some("[]")], Some("[{}]")),
+            (
+                vec![Some(r#"{}"#), Some(r#"[]"#), Some(r#"3"#), Some(r#""4""#)],
+                Some(r#"[{}, 3, "4"]"#),
+            ),
+            (
+                vec![Some("[1, 2]"), Some("[3, 4]")],
+                Some(r#"[1, 2, 3, 4]"#),
+            ),
+        ];
+
+        for (vargs, expected) in cases {
+            let vargs = vargs
+                .into_iter()
+                .map(|input| input.map(|s| Json::from_str(s).unwrap()))
+                .collect::<Vec<_>>();
+            let expected = expected.map(|s| Json::from_str(s).unwrap());
+
+            let output = RpnFnScalarEvaluator::new()
+                .push_params(vargs.clone())
+                .evaluate(ScalarFuncSig::JsonMergeSig)
+                .unwrap();
+            assert_eq!(output, expected, "{:?}", vargs);
+        }
+    }
+
+    #[test]
+    fn test_json_object() {
+        let cases = vec![
+            (vec![], r#"{}"#),
+            (vec![("1", None)], r#"{"1":null}"#),
+            (
+                vec![
+                    ("1", None),
+                    ("2", Some(r#""sdf""#)),
+                    ("k1", Some(r#""v1""#)),
+                ],
+                r#"{"1":null,"2":"sdf","k1":"v1"}"#,
+            ),
+        ];
+
+        for (vargs, expected) in cases {
+            let vargs = vargs
+                .into_iter()
+                .map(|(key, value)| (Bytes::from(key), value.map(|s| Json::from_str(s).unwrap())))
+                .collect::<Vec<_>>();
+
+            let mut new_vargs: Vec<ScalarValue> = vec![];
+            for (key, value) in vargs.into_iter() {
+                new_vargs.push(ScalarValue::from(key.clone()));
```

Comment:
`(key, value)` seems just being used in the loop just one time. Could we remove `.clone()`?

Reply:
> `(key, value)` seems just being used in the loop just one time. Could we remove `.clone()`?

Yes, I'll change it.